myriad-scheduler/src/main/java/org/apache/myriad/scheduler/NMExecutorCommandLineGenerator.java

```diff
@@ -0,0 +1,151 @@
+    CommandInfo.Builder builder = CommandInfo.newBuilder();
+    StringBuilder cmdLine = new StringBuilder();
+    appendCgroupsCmds(cmdLine);
+    appendDistroExtractionCommands(cmdLine);
+    appendUserSudo(cmdLine);
+    cmdLine.append(YARN_NM_CMD);
+    builder.setValue(String.format(CMD_FORMAT, cmdLine.toString()));
+    builder.addAllUris(getUris());
+    staticCommandInfo = builder.build();
+  }
+
+  protected Protos.Environment generateEnvironment(ServiceResourceProfile profile, Collection<Long> ports) {
+    Map<String, String> yarnEnv = myriadConfiguration.getYarnEnvironment();
+    Protos.Environment.Builder builder = Protos.Environment.newBuilder();
+    builder.addAllVariables(Iterables.transform(yarnEnv.entrySet(), new Function<Map.Entry<String, String>, Protos.Environment.Variable>() {
+      public Protos.Environment.Variable apply(Map.Entry<String, String> x) {
+        return Protos.Environment.Variable.newBuilder().setName(x.getKey()).setValue(x.getValue()).build();
+      }
+    }));
+
+    StringBuilder yarnOpts = new StringBuilder();
+    String rmHostName = System.getProperty(KEY_YARN_RM_HOSTNAME);
+
+
+    if (StringUtils.isNotEmpty(rmHostName)) {
+      addJavaOpt(yarnOpts, KEY_YARN_RM_HOSTNAME, rmHostName);
+    }
+
+    if (yarnEnv.containsKey(KEY_YARN_HOME)) {
+      addJavaOpt(yarnOpts, KEY_YARN_HOME, yarnEnv.get("YARN_HOME"));
+    }
+
+    addJavaOpt(yarnOpts, KEY_YARN_NM_LCE_CGROUPS_HIERARCHY, "mesos/$TASK_DIR");
+    addJavaOpt(yarnOpts, KEY_NM_RESOURCE_CPU_VCORES, Integer.toString(profile.getCpus().intValue()));
+    addJavaOpt(yarnOpts, KEY_NM_RESOURCE_MEM_MB, Integer.toString(profile.getMemory().intValue()));
+    Map<String, Long> portsMap = profile.getPorts();
+    Preconditions.checkState(portsMap.size() == ports.size());
+
+    Iterator itr = ports.iterator();
+    for (String portProperty : portsMap.keySet()) {
+      if (portProperty.endsWith("address")) {
+        addJavaOpt(yarnOpts, portProperty, ALL_LOCAL_IPV4ADDR + itr.next());
+      } else if (portProperty.endsWith("port")) {
+        addJavaOpt(yarnOpts, portProperty, itr.next().toString());
+      } else {
+        LOGGER.warn("{} propery isn't an address or port!", portProperty);
+      }
+    }
+
+
+    if (myriadConfiguration.getYarnEnvironment().containsKey(ENV_YARN_NODEMANAGER_OPTS)) {
+      yarnOpts.append(" ").append(yarnEnv.get(ENV_YARN_NODEMANAGER_OPTS));
+    }
+    builder.addAllVariables(Collections.singleton(
+            Protos.Environment.Variable.newBuilder()
+                .setName(ENV_YARN_NODEMANAGER_OPTS)
+                .setValue(yarnOpts.toString()).build())
+    );
+    return builder.build();
+  }
+
+  protected void appendCgroupsCmds(StringBuilder cmdLine) {
+    //These can't be set in the environment as they require commands to be run on the host
+    if (myriadConfiguration.getFrameworkSuperUser().isPresent() && myriadConfiguration.isCgroupEnabled()) {
+      cmdLine.append(" export TASK_DIR=`basename $PWD`&&");
```

Comment:
can we change this to `cat /proc/self/cgroup | grep cpu ...` as discussed?  This we add to the number of systems that support cgroups.